src/image.rs

```diff
@@ -638,6 +638,7 @@ pub trait GenericImage: GenericImageView {
     fn inner_mut(&mut self) -> &mut Self::InnerImage;
 
     /// Returns a subimage that is a view into this image.
+    /// If you want an immutable subimage instead, use `GenericImageView::view`
```

Comment:
This addresses only parts of the missing documentation. The other way around may also be helpful together with a short paragraph on these methods in `SubImage`. Potentially similar to [`std::iter::Map`](https://doc.rust-lang.org/std/iter/struct.Map.html) but with a clear indication that these are helpers and `SubImage::new` is also available.

Reply:
Also, these should be proper links, not only code markup.